frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/multiple-storageclass-selection.spec.ts

```diff
@@ -59,94 +47,103 @@ describe('Add capacity using multiple storage classes', () => {
     cy.exec('oc get nodes -o json').then((res) => {
       const nodes = JSON.parse(res.stdout);
       const allWorkerNodes = nodes.items.filter((node) => getNodeRole(node) === 'worker');
-      testPersistentVolume1.spec.nodeAffinity.required.nodeSelectorTerms[0].matchExpressions[0].values[0] =
-        allWorkerNodes[0].metadata.name;
-      testPersistentVolume2.spec.nodeAffinity.required.nodeSelectorTerms[0].matchExpressions[0].values[0] =
-        allWorkerNodes[1].metadata.name;
-      testPersistentVolume3.spec.nodeAffinity.required.nodeSelectorTerms[0].matchExpressions[0].values[0] =
-        allWorkerNodes[2].metadata.name;
-      cy.exec(`echo '${JSON.stringify(testPersistentVolume1)}' | kubectl apply -f -`);
-      cy.exec(`echo '${JSON.stringify(testPersistentVolume2)}' | kubectl apply -f -`);
-      cy.exec(`echo '${JSON.stringify(testPersistentVolume3)}' | kubectl apply -f -`);
+      const { name: scName } = testNoProvisionerSC.metadata;
+      allWorkerNodes.forEach((node, id) => {
+        const nodeName = node.metadata.name;
+        cy.exec(`echo '${JSON.stringify(getTestPV(id, nodeName, scName))}' | kubectl apply -f -`);
+      });
     });
     commonFlows.navigateToOCS();
     cy.byLegacyTestID('horizontal-link-Storage Cluster').click();
   });
+
+  beforeEach(() => {
+    fetchStorageClusterJson().then((res) => {
+      const json: K8sResourceKind = JSON.parse(res.stdout);
+      beforeCapacityAddition.deviceSets = json.spec.storageDeviceSets;
+      beforeCapacityAddition.uid = json.metadata.uid;
+    });
+  });
+
   after(() => {
     cy.exec(`echo '${JSON.stringify(testEbsSC)}' | kubectl delete -f -`);
     cy.exec(`echo '${JSON.stringify(testNoProvisionerSC)}' | kubectl delete -f -`);
-    cy.exec(`echo '${JSON.stringify(testPersistentVolume1)}' | kubectl delete -f -`);
-    cy.exec(`echo '${JSON.stringify(testPersistentVolume2)}' | kubectl delete -f -`);
-    cy.exec(`echo '${JSON.stringify(testPersistentVolume3)}' | kubectl delete -f -`);
+    cy.exec('oc get nodes -o json').then((res) => {
```

Comment:
```suggestion
    cy.exec('oc get nodes -l "node-role.kubernetes.io/worker" -o json').then((res) => {
```
Directly get worker nodes.